browser/ui/webui/brave_vpn/panel_handler/vpn_panel_handler.cc

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/webui/brave_vpn/panel_handler/vpn_panel_handler.h"
+
+#include <utility>
+
+VPNPanelHandler::VPNPanelHandler(
+    mojo::PendingReceiver<brave_vpn::mojom::PanelHandler> receiver,
+    ui::MojoBubbleWebUIController* webui_controller,
+    GetWebContentsForTabCallback get_web_contents_for_tab,
+    BraveVpnService* service)
+    : receiver_(this, std::move(receiver)),
+      webui_controller_(webui_controller),
+      get_web_contents_for_tab_(std::move(get_web_contents_for_tab)),
+      service_(service) {}
+
+VPNPanelHandler::~VPNPanelHandler() {}
+
+void VPNPanelHandler::ShowUI() {
```

Comment:
AFAIK this is part of the 2 step process to show the UI and is present on all WebUI Bubbles.

1. Button is clicked and WebUI class is created - rendering begins.
2. Rendering finishes, JS, HTML & CSS layout completes and JS calls `ShowUI()` at which point the panel is shown.

This is so that layout "jank" and the flash of webui white background is avoided.

Reply:
@petemill Thanks for the explanation!